data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java

```diff
@@ -167,7 +168,13 @@ public AbstractVerticle get(final DataPlaneContract.Resource resource, final Dat
           egressSubscriberSender,
           egressDeadLetterSender,
           new HttpSinkResponseHandler(vertx, resource.getTopics(0), producer),
-          getOffsetManager(deliveryOrder, consumer, eventsSentCounter::increment)
+          getOffsetManager(deliveryOrder, consumer, eventsSentCounter::increment),
+          ConsumerTracer.create(
+            ((VertxInternal) vertx).tracer(),
+            new KafkaClientOptions()
+              .setConfig(consumerConfigs)
+              .setTracingPolicy(TracingPolicy.PROPAGATE)
+          )
```

Comment:
I don't get why we need this tracer, vertx Kafka already creates a tracer when we create a Kafka client at line 200, and it does a similar thing of https://github.com/knative-sandbox/eventing-kafka-broker/pull/900/files#diff-384d435208823cfbd29027fa5bba51ebbc6c3671ffecb12f0af13405ed9c11b1R138. 
Can you add a comment on why we set tracing policy to ignore when we create a consumer and we create our own tracer?

Reply:
Is it because it doesn't wait for our handler to finish before finishing the span and our handler is async?

Reply:
> Can you add a comment on why we set tracing policy to ignore when we create a consumer and we create our own tracer?

I was planning to clean up a bit that code and explain why we do that and also "hide" that detail a bit, maybe behind an helper. In short, the reason is both the one you sad, which makes spans looks weird and sometimes even no-sense, and this one: https://github.com/knative-sandbox/eventing-kafka-broker/issues/929

Reply:
/lgtm

Reply:
Just tried, there is not much I can hide, it will look weird anyway... I'll check out what can be done on vertx-kafka-client side to make this look better...

Reply:
Neverthless, I've made a PR to add some comments: https://github.com/knative-sandbox/eventing-kafka-broker/pull/964